File: 프로그래머스/lv2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) { 
    const answer = [0]
    const days = []
    progresses.forEach((key, idx)=>{ 
        days[idx] = Math.ceil((100 - key) / speeds[idx]);
    })  
    let num = days[0];
    for(let i=0,j=0 ; i < days.length; i++){
        if(num < days[i]){
            num = days[i]
            answer[++j] = 1
        } else {
            answer[j] += 1
        } 
        console.log(num,days[i],answer[j])
    }
     
    return answer;
}

File: 프로그래머스/lv2/기능개발.py
from collections import deque
dq = deque()

def solution(progresses, speeds):
    answer = []
    days = deque()
    tmp = 1
    for i in range(len(progresses)):
        days.append((100 - progresses[i])//speeds[i]+1)
    print('----day : ',days)
    # for i in range(len(days)):
    #     cur = days.popleft()
    #     if cur > tmp:
    #         answer[i] += 1
    #         tmp = cur
    #     else:
    #         answer.append(1)
    
    return answer

        

# i번째 작업보다 i+1번째 작업의 개발시간이 크다면 선행한 작업을 바로 배포한다.
# 만약 i+1번째 작업의 개발시간이 적다면 선행 배포시 함께 배포한다.
# 테케 추가 [90, 90, 90, 90],[30, 1, 1, 1],[1, 3]
# 테케#2 추가 [95, 95, 90, 99, 99, 80, 99] [1, 1, 1, 1, 1, 1, 1] [1, 2, 2, 2]